homeassistant/components/splunk/__init__.py

```diff
@@ -30,10 +32,26 @@
         vol.Optional(CONF_SSL, default=False): cv.boolean,
         vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
     }),
 }, extra=vol.ALLOW_EXTRA)
 
 
+def post_request(event_collector, body, headers, verify_ssl):
+    """Post request to Splunk."""
+    try:
+        payload = {
+            "host": event_collector,
+            "event": body,
+        }
+        requests.post(event_collector,
```

Comment:
Please extract the service specific interface code, I/O and data parsing, to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_component_code_review.html#4-communication-with-devices-services

Reply:
The I/O and data parsing of this component/integration is not my code. The only reason this is showing change is because I moved existing code to a function. If this is something that is required, why wasn’t it done when the component was first added?

Truly, my core change was like 3-4 lines of code. I’ve written more changes just to make it possible to patch for unit testing. I don’t know the specifics of the I/O or data parsing. I only added a minor trigger for the filtering and pulled in already built filtering features existing in HA (taken from HomeKit component).

Reply:
We're more strict today on this rule. We shouldn't make any changes to this integration until the interface code has been moved to a library.

Reply:
Well, that's quite unfortunate.  Would have been nice if someone said that during one of the first two PR's I had submitted when trying to get these mere 5 lines of code changed (most time spent figuring out how to make the unit tests).  Could have saved me from wasting a chunk of my time in an effort to help and contribute.  

If the non-compliance of the old code is a deal breaker for me being able to make a very simple change, then we can just close this PR I guess.  I'll just use my changes myself.  

I do recommend that if there is certain existing code that is going to be "off limits" for change until the existing code passes other requirements, it might be good to have a way to indicate that to those coming and wanting to help.  It is a bit frustrating to have my time wasted, after jumping through all the hoops to comply with all the various requirements... especially after finally seeing the checks complete successfully.

Reply:
I agree. I'm sorry we didn't catch this earlier.

@balloob you said the earlier PR was good to merge once tests were added. Do you want to accept this one?

Reply:
I suggest we can let this one slip.

Reply:
Thanks.  Appreciate it. I will benefit from my change either way, but I feel better about the time put in if it is extended to others.  :)